File: java/NeetCode150/7_product_except_self.java
class Solution {
    public int[] productExceptSelf(int[] nums) 
    {
        int zero_count = 0;
        int product = 1;
        for(int i=0;i<nums.length;i++)
        {
            if(nums[i] == 0)
            {
                zero_count++;
            }
            else
            {
                product = product * nums[i];
            }
        }
        int[] output = new int[nums.length];
        for(int i=0; i<nums.length;i++)
        {
            if(zero_count==0)
            {
                output[i] = product / nums[i];
            }
            else if(zero_count==1)
            {
                if(nums[i] == 0)
                {
                    output[i] = product;
                }
                else
                {
                    output[i] = 0;
                }
            }
            else if(zero_count>=2)
            {
                output[i] = 0;
            }
        }
        return output;
    }
}  

File: python/NeetCode150/7_product_except_self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        product = 1
        zero_counter = 0
        for num in nums:
            if num == 0:
                zero_counter+=1
                num = 1
            product = product * num
        output = []
        for num in nums:
            if zero_counter == 0:
                output.append(int(product/num))
            if zero_counter == 1:
                if num == 0:
                    num = 1
                    output.append(int(product/num))
                else:
                    output.append(0)
            elif zero_counter >= 2:
                output.append(0)

        return output